From 34a71972d75fdac938371177065df0f1847c1c53 Mon Sep 17 00:00:00 2001
From: Dominic Hargreaves <dom@earth.li>
Date: Sat, 28 May 2016 15:51:53 +0100
Subject: ext/POSIX/t/sigaction.t: Skip uid and pid tests on GNU/Hurd

SA_SIGINFO does not include uid and pid on GNU/Hurd, and this is
unlikely to change in the near future, so skip these tests.

Thanks to Samuel Thibault for the report.

Origin: http://perl5.git.perl.org/perl.git/commit/d54f4edc5d7b2d0c486a36819c348aaec29b8470
Bug-Debian: https://bugs.debian.org/825016
Patch-Name: fixes/hurd_sigaction.diff
---
 ext/POSIX/t/sigaction.t | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ext/POSIX/t/sigaction.t b/ext/POSIX/t/sigaction.t
index d19341c..bce3b76 100644
--- a/ext/POSIX/t/sigaction.t
+++ b/ext/POSIX/t/sigaction.t
@@ -202,7 +202,9 @@ SKIP: {
     $skip{pid}{$^O} = $skip{uid}{$^O} = "not set for kill()"
         if (($^O.$Config{osvers}) =~ /^darwin[0-8]\./
             ||
-            ($^O.$Config{osvers}) =~ /^openbsd[0-5]\./);
+            ($^O.$Config{osvers}) =~ /^openbsd[0-5]\./
+            ||
+            ($^O eq 'gnu'));
     my $tests = keys %{{ %siginfo, %opt_val }};
     eval 'use POSIX qw(SA_SIGINFO); SA_SIGINFO';
     skip("no SA_SIGINFO", $tests) if $@;
